DIR_OUTLINE.md

```diff
@@ -0,0 +1,34 @@
+**Backend**
+  - `endpoints`: Endpoints are functions used to send information to the frontend, in other words they handle HTTP requests. This directory contains functions that the endpoint functions for `post` and `get`. The get endpoints is where you send get requests to, and the post endpoint is where you send the post requests to. 
+  - `globals`: Contains the configurations for firebase
+  - `helpers`:This contains all of the functions used in the backend, for example add_donation, to help the endpoint add a donation
+  - `types`: contains the structs (similar to classes) of donation and user-data
+  - `.env.template`: the configuration for starting the firebase project
+  - `.gitignore`: contains files for github to ignore such as sensitive information like our project key
+  - `Dockerfile`: script to assemble a Docker image
+  - `compose.yml`: This is a Docker Compose file. It's used to define and run multi-container Docker applications. It uses YAML syntax to describe the services, networks, and volumes for a complete application stack.
+  - `firestore.go`: contains functions and interfaces for interacting with firestore
+  - `go.mod`: This is the Go module file. It's automatically generated by the Go tooling and it describes the module's dependencies. It allows for versioned dependencies and reproducible builds.
+  - `go.sum`: This is the Go checksum file. It's also automatically generated by the Go tooling. It contains the expected cryptographic checksums of the content of specific module versions. It's used to ensure that these modules have not been tampered with.
+  - `main.go`: Initilizes the log objects, and the routes
+  - `main_test.go`: Contains testing functions for the backend functions to make sure they work as expected 
+
+  **Frontend**
+  - `_tests_`: test for the frontend
+  - `components`: containsn the components in used, for example the `DonationCard`
+  - `cypress`: This directory contains end-to-end tests written with Cypress.
+  - `lib`: This directory contains library code / utility functions
+  - `media`: Contains media files like images or videos
+  - `pages` contains the pages of our website, for example the `donationList` page, or the home page
+  - `public`: static files that can be accessed directly by the browser.
+  - `styles`: contains the styling for our website
+  - `config`: files: Used to configure the frontend, for example configuring Sentry for error logging
+  - `middleware`: Handles the middleware of the frontend. It's used to handle requests to paths that match the pattern '/__/auth/:path'
+ 
+ `.gitignore`:
+  - Contains files that should be ignored by git in the project level, only one file is included - vercel
+  `Licence`:
+    - The Licence for our project, protecing us from copyright or plagiarism. The project uses the GNU General Public License v3.
+    
+ `*Markdown files (with md at the end)`:
```

Comment:
the title itself shouldn't have code formatting, but instead the references to markdown files inside of your text